### PR TITLE
S05 us152 specializations

### PR DESCRIPTION
**Pull request by boubseb**

### CODE REVIEW

File: src/main/java/com/example/demo/entity/Job.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    public Set<Specialization> getSpecializations() {
+        return specializations;
+    }
+
+    public void setSpecializations(Set<Specialization> specializations) {
+        this.specializations = specializations;
+    }
+
+ 
+
```
Comment (OCR-VH-DevWeb) [inline src/main/java/com/example/demo/entity/Job.java]: Think to delete the empty lines ^^

**Review by OCR-VH-DevWeb — APPROVED** [review]
Approved

File: src/main/java/com/example/demo/controller/JobController.java
Comment (Benja-san) [inline src/main/java/com/example/demo/controller/JobController.java:1]: why job controller is in the specializations us ?

File: src/main/java/com/example/demo/entity/Job.java (hunk 1 above)
Comment (Benja-san) [reply src/main/java/com/example/demo/entity/Job.java]: indent properly (for each files)

File: src/main/java/com/example/demo/entity/Specialization.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        name = "specialization_job",
+        joinColumns = @JoinColumn(name = "specializationId"),
+        inverseJoinColumns = @JoinColumn(name = "jobId")
+    )
+    private Set<Job> jobs;
+
+    @OneToMany(mappedBy = "specialization",cascade = CascadeType.ALL, orphanRemoval = true)
+    @JsonManagedReference("specialization-specializationSkill")
+    @JsonProperty("skills")
+    private List<SpecializationSkill> specializationSkill;
```
Comment (Benja-san) [inline src/main/java/com/example/demo/entity/Specialization.java]: add an s if you retrieve a list

File: src/main/java/com/example/demo/entity/SpecializationSkill.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+
+
+
+    public Integer getId() {
+        return id;
+    }
+
+    public void setId(Integer id) {
```
Comment (Benja-san) [inline src/main/java/com/example/demo/entity/SpecializationSkill.java]: remove setter for ids

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/demo/controller/PrimordialSecretController.java [hunk 4]
```diff
@@ -19,6 +20,7 @@ public class PrimordialSecretController {
 
     @Autowired
     private PrimordialSecretService primordialSecretService;
+    private AnecdoticSecretService anecdoticSecretService;
```
Comment (Benja-san) [inline src/main/java/com/example/demo/controller/PrimordialSecretController.java:23]: why using this service if you never call it ?

**Review by Benja-san — APPROVED** [review]

**Pull request merged**